### ISSUE TITLE
Swift Compiler Error for iOS 7

### ISSUE THREAD

**Issue by usami-k** [kzaher/RxSwift#74]
I'm trying to build RxSwift and RxCocoa for iOS 7.
As iOS 7 doesn't support dynamic frameworks, I added the .swift, .m, .h files to my project.
(RxSwift, RxCocoa/RxCocoa.h, RxCocoa/Common, RxCocoa/iOS)

![project](https://cloud.githubusercontent.com/assets/536781/8950556/7bed4d06-35fe-11e5-85e4-f91cffc166d4.png)

And run in RxSwift directory

```
find . -name "*.swift" -exec sed -i '' 's|^import RxSwift||' {} \;
```

But I see build error for RxCocoa...

![buildmessage](https://cloud.githubusercontent.com/assets/536781/8950624/3bef5d92-35ff-11e5-9626-6e62946b1b6a.png)

Any solution? Thank you.

**Comment by kzaher** [comment]
Hi,
we haven't officially supported iOS 7 before because of lack of time and swift compiler being pushed out of it's comfort limit with using generics in this project.

On the other hand, I've just closed this one https://github.com/kzaher/RxSwift/pull/75 with awesome work by @shuoli84.

I think we are getting more help from community now and more requests for iOS7 support, so I'll try to do something about this. I won't lie to you, swift compiler freaks me out, and because of it I can't make any guarantees this will work seamlessly.

RxCocoa has 1% objective C code because it has to interact closely with Objective-C APIs, and swift doesn't have the dynamic dispatch capabilities needed.

I've just managed to compile everything for iOS 7, including RxCocoa. Here is a rough list of the things you need to do.

* copy files
* remove `import RxSwift` like you've done
* Include ObjC bridging header: Build Settings > Swift Compiler - Code Generation > Objective-C Bridging Header = [Path to RxCocoa files]/RxCocoa/RxCocoa.h
* remove a couple of duplicate definitions from RxCocoa.swift like

```
func removingObserverFailed() {
    rxFatalError("Removing observer for key failed")
}

func handleVoidObserverResult(result: RxResult<Void>) {
    handleObserverResult(result)
}

func rxFatalError(lastMessage: String) {
    // The temptation to comment this line is great, but please don't, it's for your own good. The choice is yours.
    fatalError(lastMessage)
}

extension NSObject {
    func rx_synchronized<T>(@noescape action: () -> T) -> T {
        objc_sync_enter(self)
        let result = action()
        objc_sync_exit(self)
        return result
    }
}
```

And that should be it.

I'll probably add preprocessor definitions and help users so they don't need to do those things manually.

But I'll need a couple of days, and some help on this.

Hope this helps.

Kruno

**Comment by usami-k** [comment]
Thank you for your instruction.
I've succeed to build RxCocoa for iOS 7.

**Comment by kzaher** [comment]
We'll try to improve that for next release.

**Comment by edopelawi** [comment]
Hi guys, is this issue already fixed on `2.3.1`?

Thanks! :)
